The job **`mingw-check`** failed! Check out the build log: [(web)](https://github.com/rust-lang/rust/actions/runs/5100107868/jobs/9168193200) [(plain)](https://github.com/rust-lang/rust/commit/abc3df68035cb25dc410950a5fa55065fa421d77/checks/13808821666/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain
     |
1579 |         buffer.into_inner().unwrap(),
     |                             ^^^^^^ method not found in `String`

error[E0599]: no method named `unwrap` found for opaque type `impl std::fmt::Display + rustc_data_structures::captures::Captures<'_> + '_` in the current scope
    --> src/librustdoc/html/render/print_item.rs:1580:51
     |
1580 |         document(cx, it, None, HeadingOffset::H2).unwrap()
     |                                                   ^^^^^^ method not found in `impl Display + Captures<'_>`
For more information about this error, try `rustc --explain E0599`.
error: could not compile `rustdoc` (lib) due to 2 previous errors
Build completed unsuccessfully in 0:01:24
```

</details>

--- a/src/librustdoc/html/render/print_item.rs
+++ b/src/librustdoc/html/render/print_item.rs
@@ -1556,21 +1556,23 @@ fn item_static(w: &mut Buffer, cx: &mut Context<'_>, it: &clean::Item, s: &clean
     write!(w, "{}", document(cx, it, None, HeadingOffset::H2))
 }
 
-fn item_foreign_type(w: &mut Buffer, cx: &mut Context<'_>, it: &clean::Item) {
-    wrap_item(w, |w| {
-        w.write_str("extern {\n");
-        render_attributes_in_code(w, it, cx.tcx());
+fn item_foreign_type(w: &mut impl fmt::Write, cx: &mut Context<'_>, it: &clean::Item) {
+    let mut buffer = Buffer::new();
+    wrap_item(&mut buffer, |buffer| {
+        buffer.write_str("extern {\n");
+        render_attributes_in_code(buffer, it, cx.tcx());
         write!(
-            w,
+            buffer,
             "    {}type {};\n}}",
             visibility_print_with_space(it.visibility(cx.tcx()), it.item_id, cx),
             it.name.unwrap(),
         );
     });
 
-    write!(w, "{}", document(cx, it, None, HeadingOffset::H2));
+    write!(w, "{}{}", buffer.into_inner(), document(cx, it, None, HeadingOffset::H2)).unwrap();
 
     write!(w, "{}", render_assoc_items(cx, it, it.item_id.expect_def_id(), AssocItemRender::All))
+        .unwrap();
 }
 
 fn item_keyword(w: &mut Buffer, cx: &mut Context<'_>, it: &clean::Item) {

